### PR TITLE
First bash at #121

### PR DESCRIPTION
**Pull request by Robinlovelace**

### CODE REVIEW

**Comment by Robinlovelace** [comment]
Finally CI passes on Ubuntu :tada:

**Comment by agila5** [comment]
Hi and thanks for the PR!

File: .github/workflows/R-CMD-check.yaml [hunk 1]
```diff
@@ -65,7 +65,7 @@ jobs:
           sysreqs=$(Rscript -e "cat(sysreqs::sysreq_commands('DESCRIPTION'))")
           sudo -s eval "$sysreqs"
           # install spatial dependencies
-          sudo add-apt-repository ppa:ubuntugis/ppa
+          sudo add-apt-repository ppa:ubuntugis/ubuntugis-unstable
```
Comment (agila5) [inline .github/workflows/R-CMD-check.yaml:68]: I know that you discussed these topics several times with Prof. Bivand on stplanr repo but I still don't understand what's going on here 😅 What's the difference between ppa and ubuntugis-unstable?

File: .github/workflows/R-CMD-check.yaml [hunk 2]
```diff
@@ -86,6 +86,8 @@ jobs:
         run: |
           remotes::install_deps(dependencies = TRUE)
           remotes::install_cran("rcmdcheck")
+          # reinstall sf matching new OSGeo packages
+          install.packages("sf", repos = "https://cloud.r-project.org")
```
Comment (agila5) [inline .github/workflows/R-CMD-check.yaml:90]: The same question as before since I don't understand why it should reinstall sf

File: R/file.R [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#' Get the location of files downloaded by osmextractr
+#'
+#' This function takes a place name and
```
Comment (agila5) [inline R/file.R:3]: Maybe you forgot to commit something here. Probably it should be something like "This function takes a place name and it returns the path of `.pbf` and `.gpkg` files associated with it" and then "details: The matching between the existing files (downloaded by osmextract and stored in `oe_download_directory()`) and the input place is performed using a regular expression with a pattern equal to the input place. If the function finds no file associated with the input `place` and `download` parameter is equal to `TRUE` (`FALSE` by default), then it tries to download and convert a new file from the chosen provider".

File: .github/workflows/R-CMD-check.yaml (hunk 1 above)
Comment (Robinlovelace) [reply .github/workflows/R-CMD-check.yaml:68]: > What's the difference between ppa and ubuntugis-unstable?

Long story short: ppa on 16.04 gives very out-of-date OSGeo libs, ubuntugis-unstable gives slightly out-of-date OSGeo libs!

File: .github/workflows/R-CMD-check.yaml (hunk 2 above)
Comment (Robinlovelace) [reply .github/workflows/R-CMD-check.yaml:90]: The binary version from the [RSPM](https://docs.rstudio.com/rspm/admin/quickstarts.html) cannot find the updated OSGeo libs installed in a previous stage. See https://github.com/r-lib/actions/issues/163 for details.

File: R/file.R (hunk 3 above)
Comment (Robinlovelace) [reply R/file.R:3]: Yes that's what I meant!

**Comment by agila5** [comment]
@Robinlovelace I modified several things here. This is the current output of the examples: 

``` r
library(osmextract)
#> Data (c) OpenStreetMap contributors, ODbL 1.0. https://www.openstreetmap.org/copyright.
#> Any product made from OpenStreetMap must cite OSM as the data source.
#> Geofabrik data are taken from https://download.geofabrik.de/
#> For usage details of bbbike data see https://download.bbbike.org/osm/

res = oe_get("ITS Leeds", provider = "test")
oe_find("ITS Leeds", provider = "test")
#> [1] "C:\\Users\\Utente\\AppData\\Local\\Temp\\RtmpcruleH/test_its-example.gpkg"   
#> [2] "C:\\Users\\Utente\\AppData\\Local\\Temp\\RtmpcruleH/test_its-example.osm.pbf"
oe_find("Isle of Wight")
#> Error: No file associated with that place name could be found.
oe_find("Isle of Wight", download_if_missing = TRUE)
#> No file associated with that place name could be found.
#> Trying to download osm data with oe_get().
#> [1] "C:\\Users\\Utente\\AppData\\Local\\Temp\\RtmpcruleH/geofabrik_isle-of-wight-latest.gpkg"   
#> [2] "C:\\Users\\Utente\\AppData\\Local\\Temp\\RtmpcruleH/geofabrik_isle-of-wight-latest.osm.pbf"
oe_find("Leeds", provider = "bbbike", download_if_missing = TRUE)
#> No file associated with that place name could be found.
#> Trying to download osm data with oe_get().
#> [1] "C:\\Users\\Utente\\AppData\\Local\\Temp\\RtmpcruleH/bbbike_Leeds.gpkg"   
#> [2] "C:\\Users\\Utente\\AppData\\Local\\Temp\\RtmpcruleH/bbbike_Leeds.osm.pbf"
```

<sup>Created on 2020-09-08 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

If you like it, feel free to merge the PR

**Comment by Robinlovelace** [comment]
Great work @agila5 it looks good to me!

**Pull request merged**